### ISSUE TITLE
C++ remote development doesn't use compile_commands

### ISSUE THREAD

**Issue by viveksjain** [facebook/nuclide#954]
I managed to get remote clang service working (`clang_server.py` couldn't find `libclang.so`, but for remote development it doesn't output a message anywhere…) with default flags, but flags from `compile_commands.json` are still not being used. How can I debug what's going on?

I'm not sure how this RPC framework works and what's executed on client vs server.

**Comment by hansonw** [comment]
Sorry, just got back from vacation. Where is your `compile_commands.json` relative to your remote project root? It needs to be either at the root or symlinked there (i.e. it can't be in a subdirectory). If you added it recently you might have to restart the server to pick it up.

You can check the server logs remotely via `$TMPDIR/nuclide-$USER-logs/nuclide.log` to try and debug.

**Comment by viveksjain** [comment]
No worries, hope you had a good break! I spent (one day of) my break trying to get nuclide works because it looks so amazing. :)

My `compile_commands.json` is in the project root (although from what I saw in the code it actually walks up recursively to try to find `compile_commands.json`). I did check `/tmp/nuclide-$USER-logs/nuclide.log` but it only has messages like
```
nuclide - ClangServer-[$FILE] - created child process with PID: [$PID]
```
which don't help diagnose this issue. I also looked at `/tmp/nuclide-$USER-logs/clang/[$FILE]-[$BLAH].log`, but that also doesn't contain any relevant messages.

Doing a `ps aux | grep clang` shows process `python2.7 /usr/lib/node_modules/nuclide/pkg/nuclide-clang-rpc/python/clang_server.py -- [$FILE] [$ARGS]` where $ARGS are my default compilation flags.

I also briefly tried adding debug log messages to my atom nuclide package as well as the npm nuclide package on the server, but didn't see anything relevant being printed (thus the question about how to debug and what's executed on client vs server). Might have been doing something wrong though.

**Comment by viveksjain** [comment]
BTW, there might be something wrong with my `compile_commands.json` since it's my first time trying to generate it – if there might be an issue there, let me know how to test the file too.

**Comment by viveksjain** [comment]
NVM, I was being silly :). The debug log messages I added weren't showing up because I was using console.log which apparently isn't printed to nuclide.log. You have to use logger (or in my case, just output to a file). Once I figured that out, debugging was pretty simple.

Turns out I had a (invalid) `compile_commands.json` in one of the higher up directories that I forgot about, which was shadowing the one at my root. Once I fixed that, and added workaround flags for #916, everything worked perfectly!

**Comment by hansonw** [comment]
Ah cool. Glad you figured it out! :)
